internal/testutils/utils.go

```diff
@@ -49,14 +49,48 @@ type TestContext struct {
 }
 
 // NewTestContext returns a TestContext containing a new kubebuilder TestContext.
-func NewTestContext(binary string, env ...string) (tc TestContext, err error) {
-	tc.TestContext, err = kbtestutils.NewTestContext(binary, env...)
+// Construct if your environment is connected to a live cluster, ex. for e2e tests.
+func NewTestContext(binaryName string, env ...string) (tc TestContext, err error) {
+	if tc.TestContext, err = kbtestutils.NewTestContext(binaryName, env...); err != nil {
+		return tc, err
+	}
 	tc.ProjectName = strings.ToLower(filepath.Base(tc.Dir))
-	tc.ImageName = fmt.Sprintf("quay.io/example/%s:v0.0.1", tc.ProjectName)
-	tc.BundleImageName = fmt.Sprintf("quay.io/example/%s-bundle:v0.0.1", tc.ProjectName)
+	tc.ImageName = makeImageName(tc.ProjectName)
+	tc.BundleImageName = makeBundleImageName(tc.ProjectName)
 	tc.isOLMManagedBySuite = true
 	tc.isPrometheusManagedBySuite = true
-	return tc, err
+	return tc, nil
+}
+
+// NewPartialTestContext returns a TestContext containing a partial kubebuilder TestContext.
+// This object needs to be populated with GVK information. The underlying TestContext is
+// created directly rather than through a constructor so cluster-based setup is skipped.
+func NewPartialTestContext(binaryName, dir string, env ...string) (tc TestContext, err error) {
```

Comment:
This constructor was needed because `kbtestutils.NewTestContext()` is for live cluster testing and uses cluster API calls to set up e2e state. The partial context returned here is useful for local, non-cluster testing because it doesn't include these calls.

Reply:
The problem shows to be the kubetcl.Version() https://github.com/kubernetes-sigs/kubebuilder/commit/ea3a94a97ad4ba0cb148e711095530286d050d54#diff-7b614da29899d7d7eb4aa3b60ed598604da518c4f2dd1e6af954c942fb1c3dc3R62 introduced in the PR: https://github.com/kubernetes-sigs/kubebuilder/pull/1840

However, it shows fine. But WDTY about rename it accordingly with the definition that you made on the comment? I mean, something such as `NewNoClusterTestContext` or `NewLocalTestContext`?

Reply:
I was thinking `Partial` to inform the caller that the context they get back won't have certain fields populated, but cluster vs. local could work too. Follow-up?